### ISSUE TITLE
✅ 🐞 Bug Report: Styles repeatedly lost after save/update until manually regenerating CSS

### ISSUE THREAD

**Issue by jamieburchell** [elementor/elementor#16591]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

## Description

When setting the margin for tablet on a post content widget, the relevant CSS is not in the stylesheet until I regenerate the CSS from the tools menu. As soon as I make any unrelated changes to the template, that same style information is lost again until I regenerate the CSS.

## Steps to reproduce

1. Set margin on text widget (desktop)
2. Set margin 0 on same widget (tablet)

Displays fine in Elementor backend, doesn't on front end and CSS is missing from generated stylesheet until CSS is regenerated manually from the tools menu.

See "[broken](https://pastebin.com/LsJwEcGE)" and "[fixed](https://pastebin.com/deBEZpUD)" (regenerated) CSS (beautified)

```
$ diff broken.css fixed.css
406a407,409
>     .elementor-57 .elementor-element.elementor-element-2f1bea4>.elementor-widget-container {
>         margin: 0px 0px 0px 0px;
>     }
429a433,435
>     }
>     .elementor-57 .elementor-element.elementor-element-483097e>.elementor-widget-container {
>         margin: 0px 0px 0px 0px;
```

Using Hello theme

## Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a default WordPress theme active.
- [X] I can reproduce this bug consistently using the steps above.

## Environment
https://pastebin.com/raw/1WpHk0hi

**Comment by jamieburchell** [comment]
Yes, but the widget affected and the steps above are standard Elementor

**Comment by brittaweller** [comment]
I have the same problem. Is a really big one in my opinion as one not always notices it as usually the desktop version is ok but the errors then occur in tablet and mobile.
I use Elementor pro with custom breakpoints (added widescreen).

**Comment by MichaelDarko** [comment]
Yes! This same exact bug is absolutely killing me. I'm not even using custom breakpoints. All experiments are enabled.

For me it's happening with mobile padding and margin. The mobile padding is fine in the editor, but is not generated on the frontend. 

Regenerating CSS fixes it, but anytime you make any change, the mobile padding is lost again.

**Comment by MichaelDarko** [comment]
Now seeing this is 3 months old issue. God help us, because this is not getting fixed any time soon.

**Comment by BaronVonDerpsalot** [comment]
+1 this is getting to be a real pain. 

Hello theme.
Elementor.
Elementor pro.
That's it.

Make any change to a post template and the page looks like it's had a stroke.
Regenerate CSS, and everything is fine again until a page change is made.

**Comment by MichaelDarko** [comment]
@BaronVonDerpsalot 
> +1 this is getting to be a real pain.

Pain is an essential part of working with Elementor. I usually spend 50% of my time working and 50% trying to work around bugs.

But don't worry, they will fix this in year or two... After introducing tons of other bugs.

#16819

**Comment by NorthTexasDigital** [comment]
I have the same issue whenever WP Engine automatically updates Elementor. It is causing client issues.

**Comment by Cheizr** [comment]
> +1 this is getting to be a real pain.
> 
> Hello theme. Elementor. Elementor pro. That's it.
> 
> Make any change to a post template and the page looks like it's had a stroke. Regenerate CSS, and everything is fine again until a page change is made.

I was trying to find for 2 days what was the bug, any time i made a change on a loop grid template it brokes, after a css regenerate it works fine again, and so on...

**Comment by tahmidbintaslim** [comment]
I also see this issue in the latest Elementor Version 3.10.2. After editing pages, CSS is not generated until manually go to Tool and Regenerate CSS

**Comment by aaronnnewman** [comment]
I had this issue, I found a solution that worked for me. I deactivated all experiences in the settings. I hope this helps someone.

**Comment by cgpotato** [comment]
Constantly having similar issues, Saving a page to a template lost all tablet/mobile custom sizing is gone.

**Comment by annitacaniglia** [comment]
Same here. Why aren't there any responses on this? It makes Custom Breakpoints totally unusable, and they are now marked as STABLE FEATURES which they obviously are not.

**Comment by jackladrera** [comment]
https://github.com/elementor/elementor/issues/4464#issuecomment-1485032570
try this one

**Comment by davidd3web** [comment]
I'm on the same boat. when I regenerate the css, I keep loosing random css files too

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core 3.15.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
